frame/support/src/storage/generator/map.rs

```diff
@@ -123,6 +123,52 @@ impl<K: FullEncode, V: FullCodec, G: StorageMap<K, V>> storage::StorageMap<K, V>
 		ret
 	}
 
+	fn mutate_exists<KeyArg: EncodeLike<K>, R, F: FnOnce(&mut Option<V>) -> R>(key: KeyArg, f: F) -> R {
+		let final_key = Self::storage_map_final_key(key);
+		let mut val = unhashed::get(final_key.as_ref());
+
+		let ret = f(&mut val);
+		match val {
+			Some(ref val) => unhashed::put(final_key.as_ref(), &val.borrow()),
```

Comment:
I see this borrow is on the above code but this is not needed, val is actually the type we want to store.
```suggestion
			Some(ref val) => unhashed::put(final_key.as_ref(), &val),
```